persistence/src/main/java/com/lightbend/lagom/javadsl/persistence/ReadSideProcessor.java

```diff
@@ -0,0 +1,118 @@
+import akka.japi.Pair;
+import akka.stream.javadsl.Flow;
+import akka.stream.javadsl.Sink;
+import org.pcollections.PSequence;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * A read side processor.
+ *
+ * Read side processors consume events produced by {@link com.lightbend.lagom.javadsl.persistence.PersistentEntity}
+ * instances, and update some read side data store that is optimized for queries.
+ *
+ * The events they consume must be tagged, and a read side is able to consume events of one or more tags. Events are
+ * usually tagged according to some supertype of event, for example, events may be tagged as <code>Order</code> events.
+ * They may also be tagged according to a hash of the ID of the entity associated with the event - this allows read
+ * side event handling to be sharded across many nodes.  Tagging is done using
+ * {@link com.lightbend.lagom.javadsl.persistence.AggregateEventTag}.
+ *
+ * Read side processors are responsible for tracking what events they have already seen. This is done using offsets,
+ * which are sequential values associated with each event. Note that end users typically will not need to handle
+ * offsets themselves, this will be provided by Lagom support specific to the read side datastore, and end users can
+ * just focus on handling the events themselves.
+ */
+public abstract class ReadSideProcessor<Event extends AggregateEvent<Event>> {
+
+    /**
+     * Return a {@link ReadSideHandler} for the given offset type.
+     *
+     * @return The offset processor.
+     */
+    public abstract ReadSideHandler<Event> buildHandler();
+
+    /**
+     * An read side offset processor.
+     *
+     * This is responsible for the actual read side handling, including handling offsets and the events themselves.
+     */
+    public static abstract class ReadSideHandler<Event extends AggregateEvent<Event>> {
+
+        /**
+         * Prepare the database for all processors.
+         *
+         * This will be invoked at system startup. It is guaranteed to only be invoked once at a time across the entire
+         * cluster, and so is safe to be used to perform actions like creating tables, that could cause problems if
+         * done from multiple nodes.
+         *
+         * It will be invoked again if it fails, and it may be invoked multiple times as nodes of the cluster go up or
+         * down. Unless the entire system is restarted, there is no way to guarantee that it will be invoked at a
+         * particular time - in particular, it should not be used for doing upgrades unless the entire system is
+         * restarted and a new cluster built from scratch.
+         *
+         * @return A completion stage that is redeemed when preparation is finished.
+         */
+        public CompletionStage<Done> globalPrepare() {
+            return CompletableFuture.completedFuture(Done.getInstance());
+        }
```

Comment:
@patriknw New global prepare method for creating tables in a serial manner across the cluster.